shared/language/extract.go

```diff
@@ -1,94 +1,75 @@
 package language
 
 import (
-	"database/sql"
 	"encoding/xml"
+	"errors"
+	"github.com/itsabot/abot/core/log"
+	"github.com/itsabot/abot/shared/datatypes"
+	"github.com/itsabot/abot/shared/helpers/address"
+	"github.com/jmoiron/sqlx"
 	"io/ioutil"
+	"math/big"
 	"net/http"
 	"net/url"
 	"os"
 	"regexp"
 	"strconv"
 	"strings"
-
-	"github.com/itsabot/abot/shared/datatypes"
-	"github.com/itsabot/abot/shared/helpers/address"
-	"github.com/itsabot/abot/core/log"
-	"github.com/jmoiron/sqlx"
 )
 
 var regexCurrency = regexp.MustCompile(`\d+\.?\d*`)
 var regexNum = regexp.MustCompile(`\d+`)
 var regexNonWords = regexp.MustCompile(`[^\w\s]`)
 
-// ExtractCurrency returns a pointer to a string to allow a user a simple check
-// to see if currency text was found. If the response is nil, no currency was
-// found. This API design also maintains consistency when we want to extract and
-// return a struct (which should be returned as a pointer).
-func ExtractCurrency(s string) sql.NullInt64 {
+var ErrNotFound error = errors.New("Couldn't extract requested type from string")
+
+// ExtractCurrency returns an int64 if a currency is found, and throws an
+// error if one isn't.
+func ExtractCurrency(s string) (int64, error) {
 	log.Debug("extracting currency")
-	n := sql.NullInt64{
-		Int64: 0,
-		Valid: false,
-	}
 	s = regexCurrency.FindString(s)
-	if len(s) == 0 {
-		return n
+	if s == "" {
+		return 0, ErrNotFound
 	}
-	val, err := strconv.ParseFloat(s, 64)
-	if err != nil {
-		return n
+	if val, err := strconv.ParseFloat(s, 64); err == nil {
```

Comment:
For consistency, we should be checking if err != nil. Since that returns, there's no need for an else block.